database-mongoose/reviews.service.js

```diff
@@ -0,0 +1,217 @@
+
+let getReviews = (productId) => {
+  const query = { productId: productId };
+  const limit = 10;
+  const sort = { isHelpfulCount: -1 };
+  return ReviewsModel.find(query).limit(limit).sort(sort)
+    .select({
+      reviewId: 1,
+      rating: 1,
+      productId: 1,
+      color: 1,
+      configuration: 1,
+      category: 1,
+      isBestSeller: 1,
+      customerId: 1,
+      customerName: 1,
+      customerCountry: 1,
+      title: 1,
+      description: 1,
+      isHelpfulCount: 1,
+      isVerifiedPurchase: 1,
+      imageUrls: 1,
+      profilePicUrl: 1,
+      reviewDate: 1
+    })
+    .catch(err => console.log(err))
+
+};
+
+// /Reviews/getReviewSummary/:productId
+
+let getReviewSummary = (productId) => {
+  let countByReviews = [];
+  //get count of customer review grouped by ratings (1,2,3,4,5)
+  return ReviewsModel.aggregate([
+    {
+      $match: { productId: { $eq: productId } }
+    },
+    {
+      $group: {
+        _id: '$rating',
+        reviewCount: { $sum: 1 }
+      }
+    }
+  ])
+    .then(result => {
+      countByReviews = result;
+      //get average rating and total count of ratings for a product
+      return ReviewsModel.aggregate(
+        [
+          {
+            $match: { productId: { $eq: productId } }
+          },
+          {
+            $group:
+            {
+              _id: '$productId',
+              avgRating: { $avg: "$rating" },
+              totalRatings: { $sum: 1 }
+            }
+          }
+        ]
+      )
+
+    })
+    .then(ratingsByCustomer => {
+      console.log('ratingsByCustomer : ', ratingsByCustomer);
+      //calculate the customer rating count in % and save data in object to return
+      var reviewSummary = {
+        averageRating: ratingsByCustomer[0].avgRating.toFixed(2),
+        totalRatings: ratingsByCustomer[0].totalRatings,
+        fiveStar: (countByReviews.find(obj => obj._id === 5) ? ((countByReviews.find(obj => obj._id === 5).reviewCount) / ratingsByCustomer[0].totalRatings * 100).toFixed(2).concat('%') : '0%'),
+        fourStar: (countByReviews.find(obj => obj._id === 4) ? ((countByReviews.find(obj => obj._id === 4).reviewCount) / ratingsByCustomer[0].totalRatings * 100).toFixed(2).concat('%') : '0%'),
+        threeStar: (countByReviews.find(obj => obj._id === 3) ? ((countByReviews.find(obj => obj._id === 3).reviewCount) / ratingsByCustomer[0].totalRatings * 100).toFixed(2).concat('%') : '0%'),
+        twoStar: (countByReviews.find(obj => obj._id === 2) ? ((countByReviews.find(obj => obj._id === 2).reviewCount) / ratingsByCustomer[0].totalRatings * 100).toFixed(2).concat('%') : '0%'),
+        oneStar: (countByReviews.find(obj => obj._id === 1) ? ((countByReviews.find(obj => obj._id === 1).reviewCount) / ratingsByCustomer[0].totalRatings * 100).toFixed(2).concat('%') : '0%')
+      };
+      console.log(reviewSummary);
+      var promiseSummary = new Promise((resolve, reject) => {
+        resolve(reviewSummary)
+      });
```

Comment:
You might want to handle your reject case here

Reply:
Great point Alek. Made note of it. Planning to take care of it for next PR.